Jetpack Connect: remove plans page for woo users

v18 opened the pull request:
This PR will skip the plans page for users connecting to Jetpack after clicking a WooCommerce Services connect button. The logic was [suggested](https://github.com/Automattic/wp-calypso/pull/15741#issuecomment-313214090) by @ebinnion, it's the same one that is used for SSO users.

He also mentioned:

> we continue to add to this, we may want to abstract it.

I propose that this change is approved (pending some discussion and work around A/B testing), and then I can spend some to see if it can be abstracted away. Suggestions welcome.

# To test
## Testing that it works
1. Install WooCommerce Services on your site
2. Deactivate Jetpack
3. Make sure Jetpack will use your local calypso install for the connection by following the instructions in P7rd6c-Me-p2
4. Go to the plugins page
5. See the WooCommerce Services connect banner, and use the button to connect to Jetpack
6. See that you are not taken through the plans page during the connection flow, and are auto-redirected

## Testing all other Jetpack flows are working 👍
- [ ] test any flow that uses this functionality already, for example any of the /jetpack/connect/store flow. See FG on "Jetpack + Calypso Testing".

# To do
I found a bug while testing this flow - there's a button that says 'Redirecting you back' and if it's clicked, it takes the user to the the plans page.
- [x] make sure the final button does not lead user to plans page

jeffstieler commented on client/jetpack-connect/authorize-form.jsx:
```diff
@@ -110,10 +110,12 @@ class JetpackConnectAuthorizeForm extends Component {
 				? <LoggedInForm
 					{ ...this.props }
 					isSSO={ this.isSSO() }
+					isWCS={ this.props.jetpackConnectAuthorize.queryObject.from === 'woocommerce-services' }
```
Let's follow the `isSSO()` pattern here and only have the logic for determining if we came from WCS in one place.

v18 replied:
👍 - updated.

jeffstieler commented on client/jetpack-connect/auth-logged-in-form.jsx:
```diff
@@ -108,6 +108,12 @@ class LoggedInForm extends Component {
 			if ( ! isRedirectingToWpAdmin && authorizeSuccess ) {
 				return this.props.goBackToWpAdmin( queryObject.redirect_after_auth );
 			}
+		} else if ( props.isWCS ) {
```
Why not add this condition to the existing `if ( props.isSSO ) {` statement since the body is the same?

v18 replied:
Updated this. Didn't put them together at first so it would be clearer the two cases are unrelated, but the comment makes it clear enough and it's not difficult to separate later.

richardmuscat commented:
I know that @toddwilkens [laid down the law here](https://github.com/Automattic/wp-calypso/pull/15741#issuecomment-315306024) but [I still struggle to see the downside](https://github.com/Automattic/wp-calypso/pull/15741#issuecomment-315306024) of running the split test. 

If you really really think a split test is a bad idea (?) then this looks good.

toddwilkens commented:
@richardmuscat I did lay down the law a bit, true. But really I was just pointing out that we can do an A/B test but that we want the default/control of that experiment to be the non-plans page version. The vast majority of people who weighed in on this have the hypothesis that it will not be effective for upselling JP AND that it will hinder Woo setup. So, I want to make sure that we have the safer/simpler version set up first.

Then we can explore doing a split test. And at the point when we are thinking of exploring a split test, I think we should step back to make sure there aren't other places that seem much more likely to be effective at promoting JP plans than this one -- this is not the only place. I'd much rather put our engineering and thinking time into the places where we have much more likelihood of success...this may or may not be it.

Also, I commented similarly on the original thread.

v18 commented:
> I thought we were planning to AB test the plans page removal

@tyxla yeh, me too. I plan to wait for the discussion to be resolved before merging (noted in my description above) 👍

v18 commented:
> Do we need to add a WCS check here? https://github.com/Automattic/wp-calypso/blob/7b04266e31536aa5faad1be2a284465783fc7e30/client/jetpack-connect/auth-logged-in-form.jsx#L165

Yes, updated. This also solves the incorrect redirect button.

In 2ca90cd79b5807fffd3e466d437bf67293859352 the user is now automatically connected in - they do not have to click a button in Calypso to connect.

Here's what the full flow looks like now (cc @toddwilkens @jeffstieler):
https://cloudup.com/ci7fK_9bi2R

I rebased this branch on master. It's good to go pending an approval about A/B testing (I'm checking in about that) & a 🚢.

v18 commented:
I removed the auto-authentication part of the flow for now. The WCS flow works as before, but without auto-authentication for now.

@tyxla would you be able to give this PR another look?

I tested the regular JP cxn flow from wp-admin, and connecting from `https://wordpress.com/jetpack/connect`, and am not quite sure what else you'd like me to test before merging. Can you elaborate on the flows and steps to take to test that you consider essential?
